src/main/java/org/spongepowered/api/Client.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api;
+
+import org.spongepowered.api.entity.living.player.ClientPlayer;
+import org.spongepowered.api.network.ServerConnection;
+import org.spongepowered.api.util.ThreadContext;
+import org.spongepowered.api.world.World;
+
+import java.net.InetSocketAddress;
+import java.nio.file.Path;
+import java.util.Optional;
+
+/**
+ * Represents the Minecraft client.
+ */
+public interface Client extends ThreadContext {
+
+    /**
+     * Gets the currently loaded world, if any.
+     *
+     * @return The current world
+     */
+    Optional<World> getLoadedWorld();
```

Comment:
I'm not sure about this method.
First of all it is not entirely sure which world this is. Is this the world the client is currently viewing or the main world of the savegame and does it include the local copy of the remote world.

Then the location i would rather move it to the LocalServer interface because it is a substate of that.